app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsController.kt

```diff
@@ -36,19 +39,24 @@ class PatientSearchResultsController @Inject constructor(
             { ui: Ui -> ui.openPatientSummaryScreen(patientUuid) }
           }
 
-  private fun registerNewPatient(events: Observable<UiEvent>): Observable<UiChange> =
-      events.ofType<PatientSearchResultRegisterNewPatient>()
-          .map { it.patientName }
-          .map { patientName ->
-            OngoingNewPatientEntry(OngoingNewPatientEntry.PersonalDetails(
-                fullName = patientName,
-                dateOfBirth = null,
-                age = null,
-                gender = null))
-          }
-          .flatMap {
-            patientRepository
-                .saveOngoingEntry(it)
-                .andThen(Observable.just({ ui: Ui -> ui.openPatientEntryScreen() }))
-          }
+  private fun registerNewPatient(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
+        .ofType<PatientSearchResultRegisterNewPatient>()
+        .map { it.searchCriteria }
+        .map(this::createOngoingEntryFromSearchCriteria)
+        .flatMap(this::saveEntryAndGoToRegisterPatientScreen)
+  }
+
+  private fun saveEntryAndGoToRegisterPatientScreen(it: OngoingNewPatientEntry): Observable<UiChange> {
```

Comment:
`ongoingNewPatientEntry` instead of `it`?

Reply:
Whoops, this was an extract method which I forgot to rename. I'll fix this.